services/httpd/limiter.go

```diff
@@ -0,0 +1,54 @@
+package httpd
+
+import (
+	"errors"
+	"net/http"
+	"sync"
+
+	"github.com/influxdata/influxdb/influxql"
+)
+
+// ErrConnectionLimitExceeded indicates the connection limit has been reached.
+var ErrConnectionLimitExceeded = errors.New("connection limit exceeded")
+
+// connectionLimiter limits the maximum number of connections.
+type connectionLimiter struct {
```

Comment:
Honestly because I had no idea that existed. @jwilder it looks like it's part of the Go project, but may be counted as a separate dependency. Are we allowed to use `net/netutil`? I think it will solve this problem much better than my solution.

Although I think closing the connection with maybe a message is probably better so we might have to write our own as @joelegasse mentioned.